presto-tpch/src/main/java/com/facebook/presto/tpch/TpchRecordSet.java

```diff
@@ -200,32 +208,35 @@ public void close()
 
         private boolean rowMatchesPredicate()
         {
-            if (!predicate.isPresent()) {
+            if (!predicate.isPresent() || predicate.get().isAll()) {
                 return true;
             }
-            return predicate.get().test(rowMap());
-        }
+            Set<ColumnHandle> predicatePushdownColumns = predicate.get().getDomains().get().keySet();
+            Map<ColumnHandle, NullableValue> rowMap = predicatePushdownColumns.stream()
+                    .collect(toImmutableMap(
+                            column -> column,
+                            column -> {
+                                TpchColumnHandle tpchColumnHandle = (TpchColumnHandle) column;
+                                Type type = tpchColumnHandle.getType();
+                                TpchColumn tpchColumn = table.getColumn(tpchColumnHandle.getColumnName());
```

Comment:
Does this work regardless of `tpch.column-naming`?

Reply:
Yes.